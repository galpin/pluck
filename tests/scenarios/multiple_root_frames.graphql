{
  launches @frame {
    id
    mission_name
    rocket {
      rocket_name
    }
  }
  landpads @frame {
    id
    full_name
    location {
      region
      latitude
      longitude
    }
  }
}
