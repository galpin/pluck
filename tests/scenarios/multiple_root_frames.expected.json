{
 "launches": [
  {
   "id": "13",
   "mission_name": "Thaicom 6",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "17",
   "mission_name": "AsiaSat 6",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "25",
   "mission_name": "OG-2 Mission 2",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "1",
   "mission_name": "FalconSat",
   "rocket.rocket_name": "Falcon 1"
  },
  {
   "id": "9",
   "mission_name": "CRS-1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "11",
   "mission_name": "CASSIOPE",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "21",
   "mission_name": "ABS-3A / Eutelsat 115W B",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "7",
   "mission_name": "COTS 1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "23",
   "mission_name": "T\u00fcrkmen\u00c4lem 52\u00b0E / MonacoSAT",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "41",
   "mission_name": "CRS-11",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "35",
   "mission_name": "Iridium NEXT Mission 1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "24",
   "mission_name": "CRS-7",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "18",
   "mission_name": "CRS-4",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "45",
   "mission_name": "CRS-12",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "59",
   "mission_name": "CRS-14",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "55",
   "mission_name": "Falcon Heavy Test Flight",
   "rocket.rocket_name": "Falcon Heavy"
  },
  {
   "id": "22",
   "mission_name": "CRS-6",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "26",
   "mission_name": "Jason 3",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "56",
   "mission_name": "Paz / Starlink Demo",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "38",
   "mission_name": "SES-10",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "44",
   "mission_name": "Intelsat 35e",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "63",
   "mission_name": "SES-12",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "4",
   "mission_name": "RatSat",
   "rocket.rocket_name": "Falcon 1"
  },
  {
   "id": "32",
   "mission_name": "CRS-9",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "40",
   "mission_name": "Inmarsat-5 F4",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "42",
   "mission_name": "BulgariaSat-1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "57",
   "mission_name": "Hispasat 30W-6",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "54",
   "mission_name": "SES-16 / GovSat-1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "62",
   "mission_name": "Iridium NEXT Mission 6",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "10",
   "mission_name": "CRS-2",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "5",
   "mission_name": "RazakSat",
   "rocket.rocket_name": "Falcon 1"
  },
  {
   "id": "6",
   "mission_name": "Falcon 9 Test Flight",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "39",
   "mission_name": "NROL-76",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "58",
   "mission_name": "Iridium NEXT Mission 5",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "60",
   "mission_name": "TESS",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "19",
   "mission_name": "CRS-5",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "27",
   "mission_name": "SES-9",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "14",
   "mission_name": "CRS-3",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "20",
   "mission_name": "DSCOVR",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "33",
   "mission_name": "JCSAT-16",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "46",
   "mission_name": "FormoSat-5",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "12",
   "mission_name": "SES-8",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "15",
   "mission_name": "OG-2 Mission 1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "36",
   "mission_name": "CRS-10",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "47",
   "mission_name": "Boeing X-37B OTV-5",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "48",
   "mission_name": "Iridium NEXT Mission 3",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "2",
   "mission_name": "DemoSat",
   "rocket.rocket_name": "Falcon 1"
  },
  {
   "id": "29",
   "mission_name": "JCSAT-2B",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "37",
   "mission_name": "EchoStar 23",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "53",
   "mission_name": "ZUMA",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "30",
   "mission_name": "Thaicom 8",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "34",
   "mission_name": "Amos-6",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "43",
   "mission_name": "Iridium NEXT Mission 2",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "49",
   "mission_name": "SES-11 / Echostar 105",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "28",
   "mission_name": "CRS-8",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "31",
   "mission_name": "ABS-2A / Eutelsat 117W B",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "65",
   "mission_name": "Telstar 19V",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "3",
   "mission_name": "Trailblazer",
   "rocket.rocket_name": "Falcon 1"
  },
  {
   "id": "8",
   "mission_name": "COTS 2",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "16",
   "mission_name": "AsiaSat 8",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "50",
   "mission_name": "KoreaSat 5A",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "51",
   "mission_name": "CRS-13",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "52",
   "mission_name": "Iridium NEXT Mission 4",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "61",
   "mission_name": "Bangabandhu-1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "66",
   "mission_name": "Iridium NEXT Mission 7",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "75",
   "mission_name": "Nusantara Satu (PSN-6) / S5 / Beresheet",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "67",
   "mission_name": "Merah Putih",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "83",
   "mission_name": "Amos-17",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "69",
   "mission_name": "SAOCOM 1A",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "74",
   "mission_name": "Iridium NEXT Mission 8",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "80",
   "mission_name": "RADARSAT Constellation",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "97",
   "mission_name": "GPS III SV03 (Columbus)",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "71",
   "mission_name": "SSO-A",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "68",
   "mission_name": "Telstar 18V",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "85",
   "mission_name": "CRS-19",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "76",
   "mission_name": "CCtCap Demo Mission 1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "81",
   "mission_name": "STP-2",
   "rocket.rocket_name": "Falcon Heavy"
  },
  {
   "id": "72",
   "mission_name": "CRS-16",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "64",
   "mission_name": "CRS-15",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "78",
   "mission_name": "CRS-17",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "94",
   "mission_name": "CCtCap Demo Mission 2",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "88",
   "mission_name": "Crew Dragon In Flight Abort Test",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "77",
   "mission_name": "ArabSat 6A",
   "rocket.rocket_name": "Falcon Heavy"
  },
  {
   "id": "73",
   "mission_name": "GPS III SV01",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "101",
   "mission_name": "SAOCOM 1B, GNOMES-1, Tyvak-0172",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "70",
   "mission_name": "Es\u2019hail 2",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "82",
   "mission_name": "CRS-18",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "86",
   "mission_name": "JCSat 18 / Kacific 1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "79",
   "mission_name": "Starlink v0.9",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "84",
   "mission_name": "Starlink 1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "87",
   "mission_name": "Starlink 2",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "91",
   "mission_name": "CRS-20",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "106",
   "mission_name": "GPS III SV04 (Sacagawea)",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "98",
   "mission_name": "ANASIS-II",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "89",
   "mission_name": "Starlink 3",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "90",
   "mission_name": "Starlink 4",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "92",
   "mission_name": "Starlink 5",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "93",
   "mission_name": "Starlink 6",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "95",
   "mission_name": "Starlink 7",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "96",
   "mission_name": "Starlink-8 & SkySat 16-18",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "99",
   "mission_name": "Starlink-9 (v1.0) & BlackSky Global 5-6",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "100",
   "mission_name": "Starlink-10 (v1.0) & SkySat 19-21",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "102",
   "mission_name": "Starlink-11 (v1.0)",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "103",
   "mission_name": "Starlink-12 (v1.0)",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "104",
   "mission_name": "Starlink-13 (v1.0)",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "105",
   "mission_name": "Starlink-14 (v1.0)",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "107",
   "mission_name": "Crew-1",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "108",
   "mission_name": "Sentinel-6 Michael Freilich",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "110",
   "mission_name": "CRS-21",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "110",
   "mission_name": "SXM-7",
   "rocket.rocket_name": "Falcon 9"
  },
  {
   "id": "109",
   "mission_name": "Starlink-15 (v1.0)",
   "rocket.rocket_name": "Falcon 9"
  }
 ],
 "landpads": [
  {
   "id": "LZ-1",
   "full_name": "Landing Zone 1",
   "location.region": "Florida",
   "location.latitude": 28.485833,
   "location.longitude": -80.544444
  },
  {
   "id": "LZ-2",
   "full_name": "Landing Zone 2",
   "location.region": "Florida",
   "location.latitude": 28.485833,
   "location.longitude": -80.544444
  },
  {
   "id": "LZ-4",
   "full_name": "Landing Zone 4",
   "location.region": "California",
   "location.latitude": 34.632989,
   "location.longitude": -120.615167
  },
  {
   "id": "OCISLY",
   "full_name": "Of Course I Still Love You",
   "location.region": "Florida",
   "location.latitude": 28.4104,
   "location.longitude": -80.6188
  },
  {
   "id": "JRTI-1",
   "full_name": "Just Read The Instructions V1",
   "location.region": "Florida",
   "location.latitude": 28.4104,
   "location.longitude": -80.6188
  },
  {
   "id": "JRTI",
   "full_name": "Just Read The Instructions",
   "location.region": "California",
   "location.latitude": 33.7291858,
   "location.longitude": -118.262015
  },
  {
   "id": "ASOG",
   "full_name": "A Shortfall of Gravitas",
   "location.region": "Florida",
   "location.latitude": 28.4104,
   "location.longitude": -80.6188
  }
 ]
}